script/signup.js

```diff
@@ -0,0 +1,105 @@
+const inputEmail = document.querySelector("#inputEmail");
+const inputPw = document.querySelector("#inputPw");
+const inputPwConfirm = document.querySelector("#inputPwConfirm");
+const emailError = document.querySelector("#email_err");
+const pwError = document.querySelector("#pw_err");
+const pwConfirmError = document.querySelector("#pwConfirm_err");
+const signupBtn = document.querySelector("#signupBtn");
+const inputPwIcon = document.querySelector("#inputPwIcon");
+const inputPwComfirmIcon = document.querySelector("#inputPwComfirmIcon");
+
+let emailValid = false;
+let passwordValid = false;
+let passwordConfirmValid = false;
+
+function checkEmail() {
+    const email_format =
+        /^([0-9a-zA-Z_\.-]+)@([0-9a-zA-Z_-]+)(\.[0-9a-zA-Z_-]+){1,2}$/;
+    if (!inputEmail.value) {
+        emailError.textContent = "이메일을 입력해주세요.";
+        inputEmail.className = "error-box";
+    } else if (!email_format.test(inputEmail.value)) {
+        emailError.textContent = "올바른 이메일 주소가 아닙니다.";
+        inputEmail.className = "error-box";
+    } else if (inputEmail.value === "test@codeit.com") {
+        emailError.textContent = "이미 사용 중인 이메일입니다.";
+    } else {
+        emailError.textContent = " ";
+        inputEmail.className = "inputbox";
+        emailValid = true;
```

Comment:
위에서 설명드린 내용(early return, 그리고 전역변수 제거) 을 적용하면 다음과 같습니다.
```suggestion
    if (!inputEmail.value) {
        emailError.textContent = "이메일을 입력해주세요.";
        inputEmail.className = "error-box";
        return false;
    }
    
      if (!email_format.test(inputEmail.value)) {
        emailError.textContent = "올바른 이메일 주소가 아닙니다.";
        inputEmail.className = "error-box";
        return false;
    }
    
      if (inputEmail.value === "test@codeit.com") {
        emailError.textContent = "이미 사용 중인 이메일입니다.";
        return false;
    } 

      emailError.textContent = " ";
      inputEmail.className = "inputbox";
      return true;
```

Reply:
e561ff35cc4fc3f862271d1dfb79b9196d91d0f0 모두 수정했습니다!!